### PR TITLE
fix(jest-prop-type-error): Enable jest-prop-type-error and resolve any issues

### LINKED ISSUE #215: Enable jest-prop-type-error and resolve any issues
# Feature
## Why is this feature being added?

Ensures components are tested with all required props being passed.

## What should your feature do?
- Remove #TODO comment in jest.config.js
- Uncomment the `setupFiles` key to activate `jest-prop-type-error` library (erroring tests when a proptype warning rears its head.
- Resolve all errors stemming from uncommenting that line.

## Helpful Info
Note the mocking in use [here](https://github.com/OperationCode/front-end/blob/master/.storybook/config.js). Most of the errors will likely be related to the `rounter` being undefined.

### PR DESCRIPTION
**Pull request by sumitparakh**
Enabling jest-prop-type-error flag failed nearly 15 test cases. Fixed it.

# Description of changes

- Removes #TODO comment in jest.config.js
- Uncomments the setupFiles key to activate jest-prop-type-error library
- Above changes failed nearly 15 test cases. This PR fixes it.

# Issue Resolved

Fixes #215

### CODE REVIEW

**Comment by kylemh** [comment]
🤖 **This is a bot** 🤖<br /><br />🎉 Deployed Storybook preview! 🎉<br /><br />Click the link at the bottom of this comment to see it live.

Built with commit 2aacaeadac79bccbd1dff7e69427c29862221560

https://deploy-preview-265--operation-code-storybook.netlify.com

File: components/AdBanner/AdBanner.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
 import OutboundLink from 'components/_common_/OutboundLink/OutboundLink';
-import MockedRouter from 'test-utils/mocks/nextRouterMock';
+import Router from 'next/router';
```
Comment (kylemh) [inline components/AdBanner/AdBanner.js]: In the cases where you imported a router, can you use `withRouter` and decorate the component instead?

Comment (kylemh) [reply components/AdBanner/AdBanner.js]: If you decorate a component with `withRouter` it then has access to router props.

```javascript
import { withRouter } from 'next/router';

class SomeComponent extends React.Component {
  render() {
    const { router } = this.props;
    return <div>{router.pathname}</div>;
  }
}

export default withRouter(SomeComponent)
```

So, you'll decorate AdBanner, and then you'll have access to `router` as a prop in that component which you'll then pass to `OutboundLink`

Comment (sumitparakh) [reply components/AdBanner/AdBanner.js]: I have used props property and fixed Route error which i was getting in my local build command (yarn build).  Now every command is working in my local except **yarn export**. Not sure why.  So i have raised an issue with next.js [here](https://github.com/zeit/next.js/issues/5637).   

Let's see if they reply something in this regard.

**Comment by kylemh** [comment]
@sumitparakh I found out it's not a good idea to be using router singletons as much as my crappy OutboundLink component wants. I'm gonna remove the router dependency from that component and then your PR should be much smaller / simpler.

Thank you for bearing with me.

**Comment by kylemh** [comment]
@sumitparakh done! Give it another go.

**Comment by sumitparakh** [comment]
@kylemh . It is done finally. :dancer:

**Pull request merged**